app/controllers/account_activations_controller.rb

```diff
@@ -2,31 +2,17 @@
 
 class AccountActivationsController < ApplicationController
   def edit
-    @type = params[:type]
-    email = URI.unescape(params[:email]).downcase
-    if @type == "artists"
-      user = Artist.find_by(email: email)
-    elsif @type == "voters"
-      user = Voter.find_by(email: email)
-    elsif @type == "admins"
-      user = Admin.find_by(email: email)
-    end
+    user = UserFinder.find_by_email!(params[:type], params[:email])
```

Comment:
This is the most extreme update I made to a controller. I definitely want to slim down other controllers but this PR is already pretty large.